packages/kbn-unified-field-list/src/utils/get_warning_message.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import type { AggregateQuery, Query } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { queryCannotBeSampled } from '@kbn/esql-utils';
+
+const FIELD_STATISTICS_LABEL = i18n.translate('unifiedFieldList.fieldStats.fieldStatisticsLabel', {
+  defaultMessage: `Field statistics are`,
+});
+
+export const FIELD_DATA_LABEL = i18n.translate('unifiedFieldList.fieldStats.fieldDataLabel', {
+  defaultMessage: `Field data is`,
+});
+
+export const getReasonIfFieldStatsUnavailableForQuery = (
+  query?: AggregateQuery | Query | { [key: string]: any },
+  label: string = FIELD_STATISTICS_LABEL
+): string | undefined => {
+  if (queryCannotBeSampled(query)) {
+    return i18n.translate('unifiedFieldList.fieldStats.notAvailableForMatchESQLQueryDescription', {
+      defaultMessage: `{label} not supported for ES|QL queries with 'MATCH' or 'QSTR' functions.`,
+      values: { label },
+    });
+  }
+};
```

Comment:
This would not work for all translations. I think we better define 2 full messages.

Reply:
Updated here [`b2b7222` (#197538)](https://github.com/elastic/kibana/pull/197538/commits/b2b7222b433bed014ea4de477f3285ac9f940542)